juju/controller.py

```diff
@@ -144,6 +149,11 @@ async def connect(self, *args, **kwargs):
                 raise ValueError('Authentication parameters are required '
                                  'if controller_name not given')
             await self._connector.connect(**kwargs)
+            info = await self.info()
+            self._connector._connection.endpoints =[
+                (e, info.results[0].cacert)
+                for e in info.results[0].addresses
+            ]
```

Comment:
I'm going to be really annoying and say, can we move this after the `if...else...`. That way we only need to have this logic once.

Reply:
sure! Totally makes sense!